### PR TITLE
Signup: make the `user-first` flow the main flow

### PR DESCRIPTION
**Pull request by mattwiebe**
The AB test introduced in #27254 produced good conversion and purchase results, so now becomes the main flow.

This flow is different than the AB test in that we were able to combine the two flows we used there into a single one.

#### Testing instructions

In a **logged-out state**, visit `/start` and go through the flow. Things to note:
* You should receive an email to confirm your email after completing the first step

In a **logged-in state** visit `/start/` and create a new site. Things to note:

* You should only see `Step 1 of 3` since the user step gets removed
* You should see a normal logged-in Masterbar

### CODE REVIEW

**Comment by bsessions85** [comment]
Tests have been updated for the new flow and can be disregarded. All passed locally. https://github.com/Automattic/wp-e2e-tests/pull/1514 should go at the same time.

**Comment by bsessions85** [comment]
This also closes #27797

**Comment by scruffian** [comment]
What's the reason for splitting the flow in two pieces like this? The user should be created as soon as the form is submitted anyway even if there are more steps in the flow.

**Comment by mattwiebe** [comment]
@scruffian The reason is that we want to create the user as soon as the user step is completed, not when the flow is done. This is so that we can do follow-up with users who do not complete the entire flow. This would have required a lot of changes to the signup framework that were outside of the scope of the test, and now unfortunately also outside of the scope of shipping this.

@michaeldcain Any further comments from the previous user-first test?

**Comment by scruffian** [comment]
> The reason is that we want to create the user as soon as the user step is completed, not when the flow is done.

This is what happens at present. If you create a flow with the user step first, and complete the step then in the console you'll see a call to `/users/new`, which creates the account. The user is also logged in behind the scenes (refresh the flow and you'll be logged in).

**Comment by mattwiebe** [comment]
> If you create a flow with the user step first, and complete the step then in the console you'll see a call to /users/new, which creates the account. The user is also logged in behind the scenes (refresh the flow and you'll be logged in).

This did not work in my early attempts on either count. I'll give it a look again.

**Comment by scruffian** [comment]
> This did not work in my early attempts on either count. I'll give it a look again.

I just tested it and it is working now. Let me know if you have any problems - happy to help...

File: client/signup/config/flows.js [hunk 1]
```diff
@@ -119,20 +118,9 @@ function filterDesignTypeInFlow( flowName, flow ) {
...
-	} else {
-		// don't allow user-first phase two when logged out, has no user step
-		if ( flowName === 'user-continue' ) {
-			flowName === 'main';
-		}
-		// logged out, in main flow, maybe enter user-first abtest
-		if ( flowName === 'main' && abtest( 'userFirstSignup' ) === 'userFirst' ) {
-			flowName = 'user-first';
-		}
+	// temporary while there are still emails out there that will take people here
```
Comment (michaeldcain) [inline client/signup/config/flows.js]: This is unnecessary if we remove the `user-continue` flow. Undefined flows redirect to `main`.

**Comment by michaeldcain** [comment]
e2e updates are here (thanks, Flow Patrol!): https://github.com/Automattic/wp-e2e-tests/pull/1514

**Comment by michaeldcain** [comment]
Tested. Logged in, logged out, tried accessing old flows, tried changing username after creating account… this is so nice.

File: client/signup/config/flows.js (hunk 1 above)
Comment (mattwiebe) [reply client/signup/config/flows.js]: The more red, the merrier. Nice thinking.

**Pull request merged**